pkg/timestream/macros.go

```diff
@@ -1,98 +1,106 @@
 package timestream
 
 import (
+	"cmp"
 	"fmt"
-	"regexp"
+	"github.com/grafana/grafana-plugin-sdk-go/experimental/errorsource"
+	"github.com/grafana/timestream-datasource/pkg/models"
+	"golang.org/x/exp/maps"
+	"slices"
 	"strings"
 	"time"
-
-	"github.com/grafana/timestream-datasource/pkg/models"
 )
 
-const timeFilter = `\$__timeFilter`
-const timeFromStr = `$__timeFrom`
-const timeToStr = `$__timeTo`
-const intervalStrAlias = `$__interval`
-const intervalStr = `$__interval_ms`
-const intervalRawStr = `$__interval_raw_ms`
-const nowStr = `$__now_ms`
-
-// WHERE time > from_unixtime(unixtime)
-// WHERE time > from_iso8601_timestamp(iso_8601_string_format)
-// WHERE time > from_milliseconds(epoch_millis)
-
-// Interpolate processes macros
-func Interpolate(query models.QueryModel, settings models.DatasourceSettings) (string, error) {
+type macroFunc func(models.QueryModel, models.DatasourceSettings) (string, error)
+
+var macroFuncs = map[string]macroFunc{
+	"timeFilter":      macroTimeFilter,
+	"timeFrom":        macroTimeFrom,
+	"timeTo":          macroTimeTo,
+	"interval":        macroInterval,
+	"interval_ms":     macroInterval,
+	"interval_raw_ms": macroIntervalRaw,
+	"now_ms":          macroNow,
+	"database":        macroDatabase,
+	"table":           macroTable,
+	"measure":         macroMeasure,
+}
 
-	txt := query.RawQuery
+var macroKeys []string
 
-	if strings.Contains(txt, "$__intervalStr") {
-		return txt, fmt.Errorf("$__intervalStr removed... use $__interval_ms")
-	}
+func init() {
+	// sort macro keys longest first, so shorter keys don't clobber longer keys
+	// they're a prefix of
+	// TODO: consider refactoring sqlutil.Interpolate to be more generic and using that
+	macroKeys = maps.Keys(macroFuncs)
+	slices.SortFunc(macroKeys, func(a, b string) int { return cmp.Compare(-len(a), -len(b)) })
```

Comment:
I like Compare over a straight comparison with `>` because it's a bit more streamlined - no if/else - but subtracting them is clever! Now to figure out which to subtract from which.....